src/WebJobs.Script/BindingExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Properties;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using NuGet.Versioning;
+
+namespace Microsoft.Azure.WebJobs.Script.BindingExtensionBundle
+{
+    internal class ExtensionBundleManager : IExtensionBundleManager
+    {
+        private readonly IEnvironment _environment;
+        private readonly IOptions<ExtensionBundleOptions> _options;
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _logger;
+
+        public ExtensionBundleManager(IEnvironment environment,
+                                      IOptions<ExtensionBundleOptions> extensionBundleOptions,
+                                      HttpClient httpClient,
```

Comment:
Given the usage pattern, this could be just managed within the scope of the download method.